exasol/slc/internal/tasks/export/export_container_to_cache_task.py

```diff
@@ -0,0 +1,247 @@
+from exasol_integration_test_docker_environment.lib.base.dependency_logger_base_task import (
+    DependencyLoggerBaseTask,
+)
+from exasol_integration_test_docker_environment.lib.base.flavor_task import (
+    FlavorBaseTask,
+)
+from exasol_integration_test_docker_environment.lib.base.still_running_logger import (
+    StillRunningLogger,
+)
+from exasol_integration_test_docker_environment.lib.config.build_config import (
+    build_config,
+)
+from exasol_integration_test_docker_environment.lib.logging.command_log_handler import (
+    CommandLogHandler,
+)
+
+from exasol.slc.internal.tasks.clean.clean_images import CleanExaslcFlavorImages
+from exasol.slc.internal.tasks.export.cache_file_parameters import CacheFileParameters
+from exasol.slc.internal.tasks.export.create_and_export_container_task import (
+    CreateAndExportContainerTask,
+)
+from exasol.slc.internal.tasks.export.export_container_parameters import (
+    CHECKSUM_ALGORITHM,
+    ExportContainerParameter,
+)
+
+
+class CheckCacheFileTask(DependencyLoggerBaseTask):
+    cache_file_path: str = luigi.Parameter()  # type: ignore
+
+    def run_task(self) -> None:
+        cache_file = Path(self.cache_file_path)
+        self.return_object(cache_file.exists())
+
+
+class TmpDirectoryMakerTask(DependencyLoggerBaseTask):
+    def run_task(self) -> None:
+        temp_directory = tempfile.mkdtemp(
+            prefix="release_archive_", dir=build_config().temporary_base_directory
+        )
+        self.return_object(temp_directory)
+
+
+class ExportContainerToCacheTask(
+    FlavorBaseTask, ExportContainerParameter, CacheFileParameters
+):
+    release_image_name: str = luigi.Parameter()  # type: ignore
+
+    def __init__(self, *args, **kwargs) -> None:
+        self._check_cache_file_future: Optional[AbstractTaskFuture] = None
+        self._tmp_directory: Optional[str] = None
+        super().__init__(*args, **kwargs)
+
+    def register_required(self):
+        self._check_cache_file_future = self.register_dependency(
+            self.create_child_task(
+                task_class=CheckCacheFileTask, cache_file_path=self.cache_file_path
+            )
+        )
+
+    def run_task(self) -> Generator[BaseTask, None, None]:
+        assert self._check_cache_file_future is not None
+        is_new: bool = self.get_values_from_future(self._check_cache_file_future)  # type: ignore
+        assert isinstance(is_new, bool)
+
+        cache_file = Path(self.cache_file_path)
+        checksum_file = Path(self.checksum_file_path)
+
+        if not is_new:
+            yield from self._export_release(
+                self.release_image_name, cache_file, checksum_file
+            )
+            is_new = True
+        else:
+            yield from self._cleanup_docker_image_if_needed()
+        self.return_object(is_new)
+
+    def cleanup_task(self, success):
+        if self._tmp_directory is not None:
+            shutil.rmtree(self._tmp_directory)
```

Comment:
this could be too late, this only happens after all tasks are done or failed. luigi has the function handle_success and handle_error

Reply:
Fixed